quantumdb-postgresql/src/main/java/io/quantumdb/core/planner/TableDataMigrator.java

```diff
@@ -157,7 +157,7 @@ private String stripEscaping(String parameterName) {
 				if (resultSet.next()) {
 					Map<String, Object> id = Maps.newHashMap();
 					for (String primaryKeyColumn : primaryKeyColumns) {
-						Object value = resultSet.getObject(primaryKeyColumn);
+						Object value = parseValue(from.getColumn(primaryKeyColumn).getType().getType(),resultSet.getObject(primaryKeyColumn).toString());
```

Comment:
It would throw an error that it could not compare the postgres type with the java type, this will parse it into the compatible type.